src/components/add.promotion.container.jsx

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright © 2018 Elastic Path Software Inc. All rights reserved.
+ *
+ * This is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import intl from 'react-intl-universal';
+import { login } from '../utils/AuthService';
+import cortexFetch from '../utils/Cortex';
+import './add.promotion.container.less';
+
+const Config = require('Config');
+
+class AddPromotionContainer extends React.Component {
+  static propTypes = {
+    onSubmittedPromotion: PropTypes.func.isRequired,
+  }
+
+  constructor() {
+    super();
+    this.state = {
+      isPromotionFormOpen: false,
+      failedPromotion: false,
+      promotionCode: '',
+      couponFormLink: 'default',
```

Comment:
what's the reason for setting this to 'default'?

Reply:
Not anymore, this may or may have not have been needed when I was using optional state properties when this was still a part of the checkout summary list component.

replace with empty string